test/e2e_env/kubernetes/meshmetric/meshmetric.go

```diff
@@ -0,0 +1,249 @@
+package meshmetric
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	mads "github.com/kumahq/kuma/api/observability/v1"
+	"github.com/kumahq/kuma/pkg/plugins/policies/meshmetric/api/v1alpha1"
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/client"
+	"github.com/kumahq/kuma/test/framework/deployments/testserver"
+	"github.com/kumahq/kuma/test/framework/envs/kubernetes"
+)
+
+func BasicMeshMetricForMesh(policyName string, mesh string) InstallFunc {
+	meshMetric := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshMetric
+metadata:
+  name: %s
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+spec:
+  targetRef:
+    kind: Mesh
+  default:
+    backends:
+      - type: Prometheus
+        prometheus: 
+          port: 8080
+          path: /metrics
+          tls:
+            mode: Disabled
+`, policyName, Config.KumaNamespace, mesh)
+	return YamlK8s(meshMetric)
+}
+
+func MeshMetricWithSpecificPrometheusBackend(policyName string, mesh string, clientId string) InstallFunc {
```

Comment:
nitpick: might be easier to understand but up to you if you want to change it

```suggestion
func MeshMetricWithSpecificPrometheusClientId(policyName string, mesh string, clientId string) InstallFunc {
```

Reply:
changed